serving/resource_model_serving_provisioned_throughput.go

```diff
@@ -0,0 +1,115 @@
+package serving
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/serving"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	defaultPtProvisionTimeout = 10 * time.Minute
+)
+
+func ResourceModelServingProvisionedThroughput() common.Resource {
+	s := common.StructToSchema(
+		serving.CreatePtEndpointRequest{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(m, "name").SetForceNew()
+			common.CustomizeSchemaPath(m, "config", "traffic_config").SetComputed()
+			common.CustomizeSchemaPath(m, "config", "served_entities", "name").SetComputed()
+			common.CustomizeSchemaPath(m, "config", "served_entities", "provisioned_model_units").SetRequired()
+			common.CustomizeSchemaPath(m, "config", "served_entities", "entity_name").SetRequired()
+			common.CustomizeSchemaPath(m, "config", "served_entities", "entity_version").SetRequired()
+
+			m["serving_endpoint_id"] = &schema.Schema{
+				Computed: true,
+				Type:     schema.TypeString,
+			}
+			return m
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e serving.CreatePtEndpointRequest
+			common.DataToStructPointer(d, s, &e)
+			wait, err := w.ServingEndpoints.CreateProvisionedThroughputEndpoint(ctx, e)
+			if err != nil {
+				return err
+			}
+			endpoint, err := wait.GetWithTimeout(d.Timeout(schema.TimeoutCreate) - deleteCallTimeout)
+			if err != nil {
+				log.Printf("[ERROR] Error waiting for serving endpoint to be created: %s", err.Error())
+				nestedErr := w.ServingEndpoints.DeleteByName(ctx, e.Name)
+				if nestedErr != nil {
+					log.Printf("[ERROR] Error cleaning up serving endpoint: %s", nestedErr.Error())
+				}
+				return err
+			}
+			d.SetId(endpoint.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			var sOrig serving.ServingEndpointDetailed
+			common.DataToStructPointer(d, s, &sOrig)
+			if err != nil {
+				return err
+			}
+			endpoint, err := w.ServingEndpoints.GetByName(ctx, d.Id())
+			if err != nil {
+				return err
+			}
+			err = common.StructToData(*endpoint, s, d)
+			if err != nil {
+				return err
+			}
+			d.Set("serving_endpoint_id", endpoint.Id)
+			return nil
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e serving.CreatePtEndpointRequest
+			common.DataToStructPointer(d, s, &e)
+			if d.HasChange("config") {
```

Comment:
What about changes for budget policy, tags, and ai_gateway?

Reply:
We have already a PR that just needs to be merged after small changes, so it makes sense to do this resource correctly from the beginning